lib/liquid/standardfilters.rb

```diff
@@ -582,8 +582,7 @@ def empty?
 
       def each
         @input.each do |e|
-          e = e.respond_to?(:to_liquid) ? e.to_liquid : e
-          e.context = @context if e.respond_to?(:context=)
+          e = @context.contextualize(e)
```

Comment:
As noted in the description, this change no longer exclude `e.respond_to?(:to_liquid)`

Reply:
I can have a closer look. Making it unconditional was making a few tests of `StandardFilters` (specifically around `map`) no longer pass since there is Proc handling in said method and `Proc` does not respond `to_liquid`.

If we wanted to make the call unconditional in isolation, I will have to either:
1. Move support for `Proc` here (similar to the current PR, extending support for all use cases)
2. Drop support for `Proc` in `StandardFilters#map` (potentially what we will want to do down the road?)

I think going with 1 is a good tradeoff for now?

Reply:
Let's assume that a Proc type is the only non-liquid value that needs to be handled here.  In that case, we can use a `if e.is_a?(Proc)` check instead of `e.respond_to?(:to_liquid)`.  We can then preserve backwards compatibility with a deprecation warning for a Proc.  Doing that here would also give a desirable deprecation warning for the `e = e.call if e.is_a?(Proc)` line in `StandardFilters#map`.

This deprecation should be communicated as deprecating support for having a Proc item in an enumerable.

On the other hand, I think we should continue supporting a Proc value in liquid Hash objects and should make sure to memoize the result of evaluating the Proc.  This means that the property lookup in `StandardFilters#map` should memoize the result of calling any Proc value.

Reply:
There are other filters that support looking properties, which should be able to just use `context.lookup_and_evaluate(item, property, raise_on_not_found: false)`.  Basically, that should be used as the helper method to consolidate Proc Hash lookup support.

That helper method could be used in the `map` filter implementation as well, although we still may want to keep the `r.is_a?(Proc) ? r.call : r` line to unconditionally evaluate the Proc even when memoization isn't supported, for backwards compatibility.